paywall/src/__tests__/middlewares/web3Middleware.test.js

```diff
@@ -287,14 +301,26 @@ describe('Web3 middleware', () => {
 
   it('it should handle transaction.updated events triggered by the web3Service', () => {
     expect.assertions(1)
+    state.transactions = {
+      [transaction.hash]: {
+        ...transaction,
+        bar: 'foo',
+      },
+    }
     const { store } = create()
-    const update = {}
+    const update = {
+      foo: 'bar',
+    }
     mockWeb3Service.emit('transaction.updated', transaction.hash, update)
     expect(store.dispatch).toHaveBeenCalledWith(
       expect.objectContaining({
         type: UPDATE_TRANSACTION,
         hash: transaction.hash,
-        update,
+        update: {
+          ...transaction,
```

Comment:
Shouldn't the logic of combining the previous transaction with the new update actually be in the reducer? This would be a lot cleaner and less error prone?

Reply:
I'll open an issue to refactor this. We will need some new actions to do this cleanly

Reply:
#2737